src/darker/tests/test_main.py

```diff
@@ -76,58 +76,59 @@ def test_isort_option_with_isort_calls_sortimports(tmpdir, run_isort, isort_args
 A_PY_BLACK_ISORT = ["import os", "import sys", "", 'print("42")', ""]
 
 A_PY_DIFF_BLACK = [
-    '--- a.py',
-    '+++ a.py',
-    '@@ -1,3 +1,4 @@',
-    ' import sys',
-    ' import os',
+    "--- a.py",
+    "+++ a.py",
+    "@@ -1,3 +1,4 @@",
+    " import sys",
+    " import os",
     "-print( '42')",
-    '+',
+    "+",
     '+print("42")',
-    '',
+    "",
 ]
 
 A_PY_DIFF_BLACK_NO_STR_NORMALIZE = [
-    '--- a.py',
-    '+++ a.py',
-    '@@ -1,3 +1,4 @@',
-    ' import sys',
-    ' import os',
+    "--- a.py",
+    "+++ a.py",
+    "@@ -1,3 +1,4 @@",
+    " import sys",
+    " import os",
     "-print( '42')",
-    '+',
+    "+",
     "+print('42')",
-    '',
+    "",
 ]
 
 A_PY_DIFF_BLACK_ISORT = [
-    '--- a.py',
-    '+++ a.py',
-    '@@ -1,3 +1,4 @@',
-    '+import os',
-    ' import sys',
-    '-import os',
+    "--- a.py",
+    "+++ a.py",
+    "@@ -1,3 +1,4 @@",
+    "+import os",
+    " import sys",
+    "-import os",
     "-print( '42')",
-    '+',
+    "+",
     '+print("42")',
-    '',
+    "",
 ]
 
 
-@pytest.mark.parametrize(
-    'enable_isort, black_args, expect',
-    [
-        (False, {}, A_PY_BLACK),
-        (True, {}, A_PY_BLACK_ISORT),
-        (False, {'skip_string_normalization': True}, A_PY_BLACK_UNNORMALIZE),
-    ],
+@pytest.mark.kwparametrize(
+    dict(enable_isort=False, black_args={}, expect=A_PY_BLACK),
+    dict(enable_isort=True, black_args={}, expect=A_PY_BLACK_ISORT),
+    dict(
+        enable_isort=False,
+        black_args={"skip_string_normalization": True},
+        expect=A_PY_BLACK_UNNORMALIZE,
+    ),
     ids=["black", "black_isort", "black_unnormalize"],
 )
 @pytest.mark.parametrize("newline", ["\n", "\r\n"], ids=["unix", "windows"])
 def test_format_edited_parts(git_repo, enable_isort, black_args, newline, expect):
     """Correct reformatting and import sorting changes are produced"""
     paths = git_repo.add({"a.py": newline, "b.py": newline}, commit="Initial commit")
     paths["a.py"].write_bytes(newline.join(A_PY).encode("ascii"))
-    paths["b.py"].write_bytes("print(42 ){newline}".encode("ascii"))
+    paths["b.py"].write_bytes(f"print(42 ){newline}".encode("ascii"))
```

Comment:
Was this working before because there isn't an assertion for the newline character?

Reply:
Even now if you remove the `f` from the f-string, the test still passes. The reason is that in the `format_edited_parts()` call, only `a.py` is submitted for reformatting.

So this test also ensures that files not mentioned in `format_edited_parts(..., <changed_files>, ...)` are ignored. And in that sense, it doesn't make any difference what we write into the modified `b.py` as long as it's some Python code that *would* require reformatting if this mechanism worked wrong and didn't ignore the file.

Reply:
Even now if you remove the `f` from the f-string, the test still passes. The reason is that in the `format_edited_parts()` call, only `a.py` is submitted for reformatting.

So this test also ensures that files not mentioned in `format_edited_parts(..., <changed_files>, ...)` are ignored. And in that sense, it doesn't make any difference what we write into the modified `b.py` as long as it's some Python code that *would* require reformatting if this mechanism worked wrong and didn't ignore the file.